libraries/standard/mqtt/include/types/iot_mqtt_common_types.h

```diff
@@ -0,0 +1,557 @@
+    IOT_MQTT_NOT_INITIALIZED = 11
+} IotMqttError_t;
+
+/**
+ * @ingroup mqtt_datatypes_enums
+ * @brief Types of MQTT operations.
+ *
+ * The function @ref mqtt_function_operationtype can be used to get an operation
+ * type's description.
+ */
+typedef enum IotMqttOperationType
+{
+    IOT_MQTT_CONNECT,           /**< Client-to-server CONNECT. */
+    IOT_MQTT_PUBLISH_TO_SERVER, /**< Client-to-server PUBLISH. */
+    IOT_MQTT_PUBACK,            /**< Client-to-server PUBACK. */
+    IOT_MQTT_SUBSCRIBE,         /**< Client-to-server SUBSCRIBE. */
+    IOT_MQTT_UNSUBSCRIBE,       /**< Client-to-server UNSUBSCRIBE. */
+    IOT_MQTT_PINGREQ,           /**< Client-to-server PINGREQ. */
+    IOT_MQTT_DISCONNECT         /**< Client-to-server DISCONNECT. */
+} IotMqttOperationType_t;
+
+/**
+ * @ingroup mqtt_datatypes_enums
+ * @brief Quality of service levels for MQTT PUBLISH messages.
+ *
+ * All MQTT PUBLISH messages, including Last Will and Testament and messages
+ * received on subscription filters, have an associated <i>Quality of Service</i>,
+ * which defines any delivery guarantees for that message.
+ * - QoS 0 messages will be delivered at most once. This is a "best effort"
+ * transmission with no retransmissions.
+ * - QoS 1 messages will be delivered at least once. See #IotMqttPublishInfo_t
+ * for the retransmission strategy this library uses to redeliver messages
+ * assumed to be lost.
+ *
+ * @attention QoS 2 is not supported by this library and should not be used.
+ */
+typedef enum IotMqttQos
+{
+    IOT_MQTT_QOS_0 = 0, /**< Delivery at most once. */
+    IOT_MQTT_QOS_1 = 1, /**< Delivery at least once. See #IotMqttPublishInfo_t for client-side retry strategy. */
+    IOT_MQTT_QOS_2 = 2  /**< Delivery exactly once. Unsupported, but enumerated for completeness. */
+} IotMqttQos_t;
+
+/**
+ * @ingroup mqtt_datatypes_enums
+ * @brief The reason that an MQTT connection (and its associated network connection)
+ * was disconnected.
+ *
+ * When an MQTT connection is closed, its associated [disconnect callback]
+ * (@ref IotMqttNetworkInfo_t::disconnectCallback) will be invoked. This type
+ * is passed inside of an #IotMqttCallbackParam_t to provide a reason for the
+ * disconnect.
+ */
+typedef enum IotMqttDisconnectReason
+{
+    IOT_MQTT_DISCONNECT_CALLED,   /**< @ref mqtt_function_disconnect was invoked. */
+    IOT_MQTT_BAD_PACKET_RECEIVED, /**< An invalid packet was received from the network. */
+    IOT_MQTT_KEEP_ALIVE_TIMEOUT   /**< Keep-alive response was not received within @ref IOT_MQTT_RESPONSE_WAIT_MS. */
+} IotMqttDisconnectReason_t;
+
+/*------------------------- MQTT parameter structs --------------------------*/
+
+#ifndef IOT_MQTT_MANAGED_API
+
+/**
+ * @brief The type used to represent network connections.
+ *
+ * For the light weight MQTT API, it is application's responsibility to define and 
+ * interpret IotNetworkConnection_t, therefore it is defined as void * here.
+ */
+    typedef void * IotNetworkConnection_t;
```

Comment:
Include `iot_config.h` at the top of this file, and you don't need to do this.

Reply:
Talked to Gordon, since iot_config.h is going to be platform and demo specific, we should able to add platform specific network connection there.